packages/client/src/components/Modals/SearchPanel.vue

```diff
@@ -294,7 +294,7 @@ export default {
       this.apply();
       let searchId;
       if (this.isEditMode) {
-        this.updateSavedSearch({
+        await this.updateSavedSearch({
```

Comment:
I think most of the issues with it not applying were race conditions with saving the data and fetching, which also explains why they're obvious on staging but not locally (latency & all).